.cirrus.yml

```
@@ -0,0 +1,49 @@
+task:
+  trigger_type: manual
+  name: E2E/Integration Tests
+  timeout_in: 45
+  container:
+    image: ubuntu:21.04
+    kvm: true # required for E2E/Integrations tests
+    cpu: 4
+    memory: 8G
+  env:
+    DEBIAN_FRONTEND: noninteractive
+    CI: true
+    FORCE_COLOR: 1 # force terminal color
+    CIRRUS_CLONE_DEPTH: 0 # Required to get app version
+    PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: 1 # disable auto-browser download
+    KEYRING: /usr/share/keyrings/nodesource.gpg
+    NODE_VERSION: node_14.x # set node version e.g: "node_16.x" for Node 16 LTS
+    DISTRO: hirsute # release codename for Ubuntu 21.04 - see "lsb_release -s -c"
+  install_deps_script: |
+    apt-get update
+    apt-get install -y --no-install-recommends ca-certificates gcc g++ curl git golang openssh-client make netcat sudo vim gpg \
+                    libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb
+    curl -sLf -o /dev/null "https://deb.nodesource.com/"$NODE_VERSION"/dists/"$DISTRO"/Release"
+    curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | gpg --dearmor | tee "$KEYRING" >/dev/null
+    echo "deb [signed-by="$KEYRING"] https://deb.nodesource.com/"$NODE_VERSION" "$DISTRO" main" > /etc/apt/sources.list.d/nodesource.list
+    echo "deb-src [signed-by="$KEYRING"] https://deb.nodesource.com/"$NODE_VERSION" "$DISTRO" main" >> /etc/apt/sources.list.d/nodesource.list
```

You're also not quoting the variables, which seems bad. (I realize that they don't actually contain any spaces.)
```suggestion
    echo "deb [signed-by=${KEYRING}] https://deb.nodesource.com/${NODE_VERSION} ${DISTRO} main" > /etc/apt/sources.list.d/nodesource.list
    echo "deb-src [signed-by=${KEYRING}] https://deb.nodesource.com/${NODE_VERSION} ${DISTRO} main" >> /etc/apt/sources.list.d/nodesource.list
```
`$DISTRO` should come from something in the distro (typically `/etc/os-release`, or if you want, `lsb_release --short --codename`).

---

The container does not have `lsb-core` package installed, that's the reason that I've added manually the distro codename.
I'll move the variable around and install `lsb-core` before get the distro codename.